soc/riscv/riscv-privileged/common/soc_common_irq.c

```diff
@@ -36,6 +36,28 @@ void z_riscv_irq_priority_set(unsigned int irq, unsigned int prio, uint32_t flag
 	riscv_clic_irq_priority_set(irq, prio, flags);
 }
 
+#elif defined(CONFIG_RISCV_CUSTOM_INTERRUPT_CONTROLLER)
```

Comment:
I still have to see a justification to have this only for the RISCV privileged spec.

Reply:
Hi @carlocaione , that's a very good question!

There is a lot of confusion about all of it. I don't quite understand the idea behind the soc/riscv/riscv-privileged directory. There is no description for the CONFIG_SOC_FAMILY_RISCV_PRIVILEGED, so I could only guess.

Please correct me if I'm wrong, but from my understanding under the term "privileged" RISC-V spec means the extended ISA to be executed only in Privileged modes (non U-mode). 

But what does it mean in the light of the presence of soc/riscv/riscv-privileged directory? That we should only put the code for the platforms supporting RISC-V Privileged ISA extensions there? Then why we have such generic code as riscv-privileged/common/idle.c, vector.S, soc_irq.S there?  Probably it's just a place for generic code for RISC-V SoC but named poorly? In that sense we could add Custom IC code there as well and then refactor the naming of the directory.

I think there is a general confusion about the idea of soc/riscv/riscv-privileged directory as well as the belonging of the code placed there. And it could be addressed with a separate change?

Please share your thoughts. What do you think?

Reply:
`CONFIG_SOC_FAMILY_RISCV_PRIVILEGED` means that the platform is following the Privileged Specification as defined in Volume 2 at https://riscv.org/technical/specifications/

Reply:
Removed dependency from SoC entirely